### DISCUSSION TITLE
Error during GenerateResourcesAndImage

### DISCUSSION THREAD

**Discussion by JordanGallegos** [actions/virtual-environments#4875] [Image configuration]
Reaching out for some help or guidance regarding an error I am receiving during the image creation process. 

See error below: 

-this error occurs right after successful storage account creation

![image](https://user-images.githubusercontent.com/28936036/149200465-6e5bd442-0805-4f7b-9d84-17bdd9ec1b81.png)

-here is the ps1 script

![image](https://user-images.githubusercontent.com/28936036/149201272-6f532cd9-4759-46b6-85c9-92f36a1bc75f.png)

I followed the installation procedures and verified all build agent requirements were met. The build agent (windows 10) I am using had successfully ran through the build image process last week. I just started receiving this error starting on 1/11/2022. I also completely removed the GitHub repo from my C: drive and re-cloned. Unfortunately, still receive the same error.

**Comment by al-cheb** [comment]
Hey, @JordanGallegos 
1. What is Az module version?
2. What is the `New-AzADServicePrincipal -DisplayName $spDisplayName` command output?

**Reply by JordanGallegos** [reply]
Thanks for the help. I resolved my issue. There is currently an issue with the Az.Resources module (version 5.2). Was able to trace the reported issue here: https://github.com/Azure/azure-powershell/issues/16764. Downgrading my Az.Resources module back to  5.1 allowed for the image creation to work.
